pkg/reconciler/route/reconcile_resources.go

```diff
@@ -123,37 +125,101 @@ func (c *Reconciler) reconcileClusterIngress(
 	return clusterIngress, err
 }
 
-func (c *Reconciler) reconcilePlaceholderService(ctx context.Context, route *v1alpha1.Route, ingress *netv1alpha1.ClusterIngress) error {
+func (c *Reconciler) deleteServices(namespace string, svcs map[string]*corev1.Service) error {
+	for _, svc := range svcs {
+		if err := c.KubeClientSet.CoreV1().Services(namespace).Delete(svc.Name, nil); err != nil {
```

Comment:
If we set up the labels properly we could potentially use `DeleteCollection` instead.

Reply:
e.g. `route = <id> AND subroute NOT IN (x, y, z)`

Reply:
Unfortunately, services doesn't have `DeleteCollection` method :(
https://github.com/knative/serving/blob/c85f2136e897cd4bee9e203b40968976f098242a/vendor/k8s.io/client-go/kubernetes/typed/core/v1/service.go#L37-L47